### PR TITLE
Update Nimbus installation doc

### PR DESCRIPTION
**Pull request by AmyHoney**

### CODE REVIEW

File: docs/install/ubuntu.rst [hunk 1]
```diff
@@ -2,4 +2,195 @@
...
+
+**Prerequisites**:
+
+To keep things simple, we are going to make the following assumptions:
+
+* You run Ubuntu 20.04(focal) or later.
+* You have at least 32GB free memory and 50GB of disk space
+* You have access to the internet for downloading the required snaps and charms.
+
+**Contents**:
```
Comment (liuqi) [inline docs/install/ubuntu.rst:15]: remove this.

File: docs/install/ubuntu.rst [hunk 2]
```diff
@@ -2,4 +2,195 @@
...
+
+Part I: Create a Nimbus Ubuntu 20.04 VM
```
Comment (liuqi) [inline docs/install/ubuntu.rst:17]: remove "Part I:"

File: docs/install/ubuntu.rst [hunk 3]
```diff
@@ -2,4 +2,195 @@
...
 ==========================
 
-`Get started with Charmed Kubeflow <https://charmed-kubeflow.io/docs/get-started-with-charmed-kubeflow>`_
+This document will introduce you to all you need to know to get started with Charmed Kubeflow on Nimbus.
+
+**Prerequisites**:
+
+To keep things simple, we are going to make the following assumptions:
+
+* You run Ubuntu 20.04(focal) or later.
```
Comment (liuqi) [inline docs/install/ubuntu.rst:11]: 20.04 (focal)

File: docs/install/ubuntu.rst [hunk 4]
```diff
@@ -2,4 +2,195 @@
...
+* You have at least 32GB free memory and 50GB of disk space
+* You have access to the internet for downloading the required snaps and charms.
+
+**Contents**:
+
+Part I: Create a Nimbus Ubuntu 20.04 VM
+=======================================
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:20]: specify code type: 
.. code-block:: shell

File: docs/install/ubuntu.rst [hunk 5]
```diff
@@ -2,4 +2,195 @@
...
+
+    # login DBC account to deloy ubuntu 20.04 vm on nimbus
+    # username/password: vmware/B1gd3m0z
+    nimbus deploy ovf atlas-ubuntu-vm-3 http://sc-prd-rdops-templates.eng.vmware.com/nimbus-templates/atlas-ubuntu-20-4/atlas-ubuntu-20-04/atlas-ubuntu-20-04.ovf --cpus=16
+
+    # login this vm, and edit /etc/environment and remove proxy related env. vars, then reboot
+ 
+
+Part II: Access Charmed Kubeflow
```
Comment (liuqi) [inline docs/install/ubuntu.rst:29]: remove "Part II:"

File: docs/install/ubuntu.rst [hunk 6]
```diff
@@ -2,4 +2,195 @@
...
+
+
+Login Nimbus VM created before
+------------------------------
+
+The journey of access Charmed Kubeflow will start from login Nimbus VM.
+
+.. code-block::
+
+    ssh -D localhost:1080 vmware@<vm_machine_public_ip>
```
Comment (liuqi) [inline docs/install/ubuntu.rst:50]: vm_machine_public_ip -> vm_ip

File: docs/install/ubuntu.rst [hunk 7]
```diff
@@ -2,4 +2,195 @@
...
+
+Please note that this tutorial is dedicated to users who aim to install Charmed Kubeflow on their own machine or on a virtual machine with direct access to the browser. In case you run it on a public cloud, the deployment process is the same, but please follow these `instructions <https://charmed-kubeflow.io/docs/dashboard>`_ for accessing the dashboard.
+
+
+Login Nimbus VM created before
+------------------------------
+
+The journey of access Charmed Kubeflow will start from login Nimbus VM.
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:48]: ditto

File: docs/install/ubuntu.rst [hunk 8]
```diff
@@ -2,4 +2,195 @@
...
+
+    ssh -D localhost:1080 vmware@<vm_machine_public_ip>
+
+
+Install and prepare MicroK8s
+------------------------------
+
+The first step on our journey is to install `MicroK8s <https://microk8s.io/>`_. MicroK8s is installed from a snap package. The published snap maintains different channels for different releases of Kubernetes.
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:58]: ditto

File: docs/install/ubuntu.rst [hunk 9]
```diff
@@ -2,4 +2,195 @@
...
+
+The first step on our journey is to install `MicroK8s <https://microk8s.io/>`_. MicroK8s is installed from a snap package. The published snap maintains different channels for different releases of Kubernetes.
+
+.. code-block:: 
+
+    sudo snap install microk8s --classic --channel=1.24/stable
+
+For MicroK8s to work without having to use sudo for every command, it creates a group called microk8s. To make it more convenient to run commands, you will add the current user to this group:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:64]: ditto

File: docs/install/ubuntu.rst [hunk 10]
```diff
@@ -2,4 +2,195 @@
...
+
+It is also useful to make sure the user has the proper access and ownership of any kubectl configuration files:
+
+.. code-block:: 
+
+    sudo chown -f -R $USER ~/.kube
+
+MicroK8s will start up as soon as it is installed. It is a completely functional Kubernetes, running with the least amount of overhead possible. However, for our purposes we will need a Kubernetes with a few more features. A lot of extra services are available as MicroK8s “add-ons” - code which is shipped with the snap and can be turned on and off when it is needed. We can now enable some of these features to get a Kubernetes where we can usefully install Kubeflow. We will add a DNS service, so the applications can find each other, storage, an ingress controller so we can access Kubeflow components and the MetalLB load balancer application. These will be enabled simply at the same time:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:77]: ditto

File: docs/install/ubuntu.rst [hunk 11]
```diff
@@ -2,4 +2,195 @@
...
+
+.. code-block:: 
+
+    microk8s enable dns storage ingress metallb:10.64.140.43-10.64.140.49
+
+You can see that we added some detail when enabling MetalLB, in this case the address pool to use. Many of the add-ons have extra configuration options, which can be found in the `MicroK8s documentation <https://microk8s.io/docs/addon-metallb>`_.
+
+It will take minimum 5 minutes for MicroK8s to install and set up these additional features. Before we do anything else, we should check that the add-ons have been enabled successfully and that MicroK8s is ready for action. We can do this by requesting the status, and supplying the --wait-ready option, which tells microk8s to finish whatever processes it is working on before returning:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:85]: ditto

File: docs/install/ubuntu.rst [hunk 12]
```diff
@@ -2,4 +2,195 @@
...
+Now we have a working Kubernetes ready, the next step is to install Juju.
+
+
+Install Juju
+------------
+
+`Juju <https://juju.is/>`_ is an operation Lifecycle manager(OLM) for clouds, bare metal or Kubernetes. We will be using it to deploy and manage the components which make up Kubeflow.
+As with MicroK8s, Juju is installed from a snap package:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:98]: ditto

File: docs/install/ubuntu.rst [hunk 13]
```diff
@@ -2,4 +2,195 @@
...
+
+    microk8s status --wait-ready
+
+Now we have a working Kubernetes ready, the next step is to install Juju.
+
+
+Install Juju
+------------
+
+`Juju <https://juju.is/>`_ is an operation Lifecycle manager(OLM) for clouds, bare metal or Kubernetes. We will be using it to deploy and manage the components which make up Kubeflow.
```
Comment (liuqi) [inline docs/install/ubuntu.rst:95]: whitespace after "manager"

File: docs/install/ubuntu.rst [hunk 14]
```diff
@@ -2,4 +2,195 @@
...
+As with MicroK8s, Juju is installed from a snap package:
+
+.. code-block::
+
+    sudo snap install juju --classic
+
+As Juju already has a built-in knowledge of MicroK8s and how it works, there is no additional setup or configuration needed. All we need to do is run the command to deploy a Juju controller to the Kubernetes we set up with MicroK8s:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:104]: ditto

File: docs/install/ubuntu.rst [hunk 15]
```diff
@@ -2,4 +2,195 @@
...
+
+    juju bootstrap microk8s
+
+The controller is Juju’s agent, running on Kubernetes, which can be used to deploy and control the components of Kubeflow.
+
+The controller can work with different ``models``, which map to namespaces in Kubernetes. You set up a specific model for Kubeflow:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:112]: ditto

File: docs/install/ubuntu.rst [hunk 16]
```diff
@@ -2,4 +2,195 @@
...
+
+That’s it for installing Juju!
+
+
+Deploying Charmed Kubeflow
+--------------------------
+
+Charmed Kubeflow is really a collection of charms. Each of these charms deploy and control one application which goes to make up Kubeflow. You can actually just install the components you want, by individually deploying the charms and relating them to each other to build up Kubeflow. The bundles are really a recipe for a particular deployment of Kubeflow, configuring and relating the applications so you end up with a working deployment with the minimum of effort.
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:126]: ditto

File: docs/install/ubuntu.rst [hunk 17]
```diff
@@ -2,4 +2,195 @@
...
+
+    juju deploy kubeflow --trust
+
+Juju will now fetch the applications and begin deploying them to the MicroK8s Kubernetes. This process can take several minutes. You can track the progress by running:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:132]: ditto

File: docs/install/ubuntu.rst [hunk 18]
```diff
@@ -2,4 +2,195 @@
...
+
+While that is going in, there are two pieces of post-install configuration which can usefully be done at this point.
+
+
+Configure the components
+------------------------
+
+For authentication and allowing access to the dashboard service, some components will need to be configured with the URL to be allowed. This depends on the underlying network provider, but for the known case of running on a local MicroK8s, we also know what the URL will be. It is configured with Juju using the following commands:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:146]: ditto

File: docs/install/ubuntu.rst [hunk 19]
```diff
@@ -2,4 +2,195 @@
...
+
+    juju config dex-auth public-url=http://10.64.140.43.nip.io
+    juju config oidc-gatekeeper public-url=http://10.64.140.43.nip.io
+
+Finding the URL: If you have a different setup for MicroK8s, or you are adapting this tutorial for a different Kubernetes, you can find the URL required by examining the IP address of the ``istio-ingressgateway`` service. For example, you can determinine this information using kubectl: ``microk8s kubectl -n kubeflow get svc istio-ingressgateway-workload -o jsonpath='{.status.loadBalancer.ingress[0].ip}'``
+
+To enable simple authentication, and set a username and password for your Kubeflow deployment, run the following commands:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:155]: ditto

File: docs/install/ubuntu.rst [hunk 20]
```diff
@@ -2,4 +2,195 @@
...
+Please note that if you are in a public cloud, follow `this guide <https://charmed-kubeflow.io/docs/dashboard>`_.
+The URL for the Kubeflow dashboard is the same as the one determined earlier for the configuration steps - in the case of a default MicroK8s install, it’s: http://10.64.140.43.nip.io
+
+From a browser on your local machine, this can be reached just by copying and pasting the URL. You should then see the dex login screen, where you should enter the username( it does say email address, but whatever string you entered to configure it will work fine) and your password from the configuration step.
+
+However, for remote deployments, or running on a virtual machine, creating a SOCKS proxy is required to access the dashboard. This can be done as follows:
+
+1. Connection to the machine using ssh with SOCKS proxy enabled through the -D 1080 parameter. As in the example below:
+
+.. code-block::
```
Comment (liuqi) [inline docs/install/ubuntu.rst:174]: ditto

File: docs/install/ubuntu.rst [hunk 21]
```diff
@@ -2,4 +2,195 @@
...
+* Login Nimbus VM created before
+* Install MicroK8s
+* Install Juju
+* Deploy Charmed Kubeflow
+* Set up 5 of the main components of Charmed Kubeflow: Notebooks, Pipelines, Katib
+
+Please note that this tutorial is dedicated to users who aim to install Charmed Kubeflow on their own machine or on a virtual machine with direct access to the browser. In case you run it on a public cloud, the deployment process is the same, but please follow these `instructions <https://charmed-kubeflow.io/docs/dashboard>`_ for accessing the dashboard.
+
+
+Login Nimbus VM created before
```
Comment (liuqi) [inline docs/install/ubuntu.rst:43]: remove "created before"

**Review by liuqi — CHANGES_REQUESTED** [review]

File: docs/install/ubuntu.rst [hunk 22]
```diff
@@ -2,4 +2,195 @@
...
+
+.. image:: ../_static/install-welcome.png
+
+Click on the “Start Setup” button. On the next screen you will be asked to create a namespace. This is just a way of keeping all the files and settings from one project in a single, easy-to-access place. You can choose any name you like…
+
+.. image:: ../_static/install-namespace.png
+
+Once you click on the “Finish” button, the Dashboard will be displayed!
+
+.. image:: ../_static/install-dashboard.png
```
Comment (liuqi) [inline docs/install/ubuntu.rst:196]: Add reference:

.. seealso::

   `Get started with Charmed Kubeflow <https://charmed-kubeflow.io/docs/get-started-with-charmed-kubeflow>`_

**Comment by AmyHoney** [comment]
> _ No description provided. _

File: docs/install/ubuntu.rst [hunk 23]
```diff
@@ -2,4 +2,191 @@
...
+
+* You run Ubuntu 20.04 (focal) or later.
+
+Create a Nimbus Ubuntu 20.04 VM
+===============================
+
+.. code-block:: shell
+
+    # login DBC account to deloy ubuntu 20.04 vm on nimbus
+    # username/password: vmware/B1gd3m0z
```
Comment (arielzhu0512) [inline docs/install/ubuntu.rst:21]: This username/password is for the deployed vm, right? Shall we make it a bit more clear, like change it to "VM username/password:xxx", or move it after the vm deploy command? (I'm afraid following the above line "login DBC account" may be a tiny little confusing...)

File: docs/install/ubuntu.rst [hunk 24]
```diff
@@ -2,4 +2,191 @@
...
+    nimbus deploy ovf atlas-ubuntu-vm-3 http://sc-prd-rdops-templates.eng.vmware.com/nimbus-templates/atlas-ubuntu-20-4/atlas-ubuntu-20-04/atlas-ubuntu-20-04.ovf --cpus=16
+
+    # login this vm, and edit /etc/environment and remove proxy related env. vars, then reboot
```
Comment (arielzhu0512) [inline docs/install/ubuntu.rst:24]: Will it be better and more clear if we directly paste an edited version of /etc/environment here?

File: docs/install/ubuntu.rst [hunk 25]
```diff
@@ -2,4 +2,191 @@
...
 
-`Get started with Charmed Kubeflow <https://charmed-kubeflow.io/docs/get-started-with-charmed-kubeflow>`_
+This document will introduce you to all you need to know to get started with Charmed Kubeflow on Nimbus.
+
+**Prerequisites**:
+
+To keep things simple, we are going to make the following assumptions:
+
+* You run Ubuntu 20.04 (focal) or later.
+
```
Comment (liuqi) [inline docs/install/ubuntu.rst:12]: plz restore the following contents which have been removed:

* You have at least 32GB free memory and 50GB of disk space
* You have access to the internet for downloading the required snaps and charms.

File: docs/install/ubuntu.rst [hunk 26]
```diff
@@ -2,4 +2,191 @@
...
 ==========================
 
-`Get started with Charmed Kubeflow <https://charmed-kubeflow.io/docs/get-started-with-charmed-kubeflow>`_
+This document will introduce you to all you need to know to get started with Charmed Kubeflow on Nimbus.
+
+**Prerequisites**:
+
+To keep things simple, we are going to make the following assumptions:
+
+* You run Ubuntu 20.04 (focal) or later.
```
Comment (arielzhu0512) [inline docs/install/ubuntu.rst:11]: I remember there were two more other prerequisites. Are they not unnecessary here?

File: docs/install/ubuntu.rst [hunk 27]
```diff
@@ -2,4 +2,191 @@
...
+
+From a browser on your local machine, this can be reached just by copying and pasting the URL. You should then see the dex login screen, where you should enter the username( it does say email address, but whatever string you entered to configure it will work fine) and your password from the configuration step.
+
+However, for remote deployments, or running on a virtual machine, creating a SOCKS proxy is required to access the dashboard. This can be done as follows:
+
+1. Connection to the machine using ssh with SOCKS proxy enabled through the -D 1080 parameter. As in the example below:
+
+.. code-block:: shell
+
+    ssh -D localhost:1080 vmware@<vm_machine_public_ip>
```
Comment (arielzhu0512) [inline docs/install/ubuntu.rst:166]: In this doc, at the very first beginning, we've already provided them with this command, with -D localhost:1080, to login VM. So is it necessary to put this command here again?

File: docs/install/ubuntu.rst (hunk 25 above)
Comment (AmyHoney) [reply docs/install/ubuntu.rst:14]: Done, plz review again.

File: docs/install/ubuntu.rst [hunk 28]
```diff
@@ -2,4 +2,191 @@
...
+To enable simple authentication, and set a username and password for your Kubeflow deployment, run the following commands:
+
+.. code-block:: shell
+
+    juju config dex-auth static-username=admin
+    juju config dex-auth static-password=admin
+
+Feel free to use a different (more secure!) password if you wish.
+
+Login to Charmed Kubeflow
```
Comment (arielzhu0512) [inline docs/install/ubuntu.rst:152]: I feel like the connections of these subsections "Deploying Charmed Kubeflow", "Configure the component", and "Login to Charmed Kubeflow" can be a bit more clear. 
At the end of subsection "Deploying Charmed Kubeflow", we write that users need to wait until everything is ready. And during the waiting process, users may take time configure the things. So at the beginning of subsection "Login to Charmed Kubeflow", shall we clearly specify something like "when everything is in "active" status, it's now time to access Kubeflow in browsers."?

**Review by arielzhu0512 — COMMENTED** [review]
Review and leave comments for install docs.

**Review by liuqi — APPROVED** [review]
I also did some more update on this docs. lgtm.

**Pull request merged**